class.jetpack.php

```diff
@@ -3018,7 +2891,7 @@ function admin_init() {
 		// If the plugin is not connected, display a connect message.
 		if (
 			// the plugin was auto-activated and needs its candy
-			Jetpack_Options::get_option( 'do_activate' )
+			Jetpack_Options::get_option_and_ensure_autoload( 'do_activate', '0' )
```

Comment:
Miguel has a fix where this function only gets call on the the `login_init` action:
https://github.com/Automattic/jetpack/pull/4271

I think that should be good enough. cc @samhotchkiss for a double check.